sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -153,6 +153,9 @@ public ValidationResult validate(boolean authorizeContextParams) throws SqlParse
     final Set<ResourceAction> resourceActions = new HashSet<>(resourceCollectorShuttle.getResourceActions());
 
     if (parsed.getInsertOrReplace() != null) {
+      if (plannerContext.getQueryContext().get(PlannerContext.CTX_SQL_OUTER_LIMIT) != null) {
```

Comment:
SuperNit: I would add a comment here explaining why it does not make sense for PlannerContext.CTX_SQL_OUTER_LIMIT to be part of Insert/Replace q.

Reply:
Added a comment